connector/ldap/ldap.go

```diff
@@ -2,20 +2,109 @@
 package ldap
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+	"log"
 
 	"gopkg.in/ldap.v2"
 
 	"github.com/coreos/dex/connector"
 )
 
-// Config holds the configuration parameters for the LDAP connector.
+// Config holds the configuration parameters for the LDAP connector. The LDAP
+// connectors require executing two queries, the first to find the user based on
+// the username and password given to the connector. The second to use the user
+// entry to search for groups.
+//
+// An example config:
+//
+//     type: ldap
+//     config:
+//       host: ldap.example.com:636
+//       rootCA: /etc/dex/ldap.ca
+//       bindDN: uid=seviceaccount,cn=users,dc=example,dc=com
+//       bindPW: password
+//       userSearch:
+//         # Would translate to the query "(&(objectClass=person)&(uid=<username>))"
+//         baseDN: cn=users,dc=example,dc=com
+//         filter: "(objectClass=person)"
+//         username: uid
+//         attributes:
+//           id: uid
+//           email: mail
+//           name: name
+//       groupSearch:
+//         # Would translate to the query "(&(objectClass=group)&(member=<user uid>))"
+//         baseDN: cn=groups,dc=example,dc=com
+//         filter: "(objectClass=group)"
+//         userAttr: uid
+//         groupAttr: member
+//         nameAttr: name
+//
 type Config struct {
-	Host   string `yaml:"host"`
+	// The host and optional port of the LDAP server. If port isn't supplied, it will be
+	// guessed based on the TLS configuration. 389 or 636.
+	Host string `yaml:"host"`
+
+	// Path to a trusted root certificate file.
+	RootCA string `yaml:"rootCA"`
+
+	// BindDN and BindPW for an application service account. The connector uses these
```

Comment:
Just realized that these can totally be empty. Some LDAP configurations allow enough anonymous access that you don't need a service account.